tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ClientCredentialsTests.cs

```diff
@@ -178,6 +179,44 @@ public async Task WithOnBeforeTokenRequest_TestAsync(Cloud cloud, TargetFramewor
             Assert.AreEqual(TokenSource.Cache, authResult.AuthenticationResultMetadata.TokenSource);            
         }
 
+        [TestMethod]
+        public async Task ByRefreshTokenTestAsync()
+        {
+            // Arrange
+            var labResponse = await LabUserHelper.GetDefaultUserAsync().ConfigureAwait(false);
+
+            var msalPublicClient = PublicClientApplicationBuilder
+                .Create(labResponse.App.AppId)
+                .WithTestLogging()
+                .WithAuthority(labResponse.Lab.Authority, "organizations")
+                .BuildConcrete();
+
+            AuthenticationResult authResult = await msalPublicClient
+                .AcquireTokenByUsernamePassword(s_scopes, labResponse.User.Upn, labResponse.User.GetOrFetchPassword())
+                .ExecuteAsync(CancellationToken.None)
+                .ConfigureAwait(false);
+
+            var confidentialApp = ConfidentialClientApplicationBuilder
+                .Create(labResponse.App.AppId)
+                .WithAuthority(labResponse.Lab.Authority, labResponse.User.TenantId)
+                .WithTestLogging()
+                .BuildConcrete();
+
+            var rt = msalPublicClient.UserTokenCacheInternal.Accessor.GetAllRefreshTokens().FirstOrDefault();
+
+            // Act
+            authResult = await (confidentialApp as IByRefreshToken).AcquireTokenByRefreshToken(s_scopes, rt.Secret).ExecuteAsync().ConfigureAwait(false);
+
+            var account = authResult.Account;
```

Comment:
The bug mentioned that `GetAccounts()` does not work. Can you use cca.GetAccount(id) nad cca.GetAccounts() ?

Reply:
@bgavrilMS  I dont see that mentioned in the issue but I will double check it.

Reply:
Irrespective of the issue, please call `GetAccount(id)` and `GetAccounts()` to ensure that those APIs work.